contracts/src/spaces/facets/roles/Roles.sol

```diff
@@ -90,7 +84,41 @@ contract Roles is IRoles, RolesBase, Entitled {
     CreateEntitlement memory entitlement
   ) external {
     _validatePermission(Permissions.ModifyRoles);
-    _checkRoleExists(roleId);
     _removeRoleFromEntitlement(roleId, entitlement);
   }
+
+  // custom channel permissions
+  function createCustomChannelPermissions(
+    uint256 roleId,
+    bytes32 channelId,
+    string[] memory permissions
+  ) external {
+    _validatePermission(Permissions.ModifyRoles);
+    _createCustomChannelPermissions(roleId, channelId, permissions);
+  }
+
+  function getCustomChannelPermissions(
+    uint256 roleId,
+    bytes32 channelId
+  ) external view returns (string[] memory permissions) {
+    return _getCustomChannelPermissions(roleId, channelId);
+  }
+
+  function updateCustomChannelPermissions(
+    uint256 roleId,
+    bytes32 channelId,
+    string[] memory permissions
+  ) external {
+    _validatePermission(Permissions.ModifyRoles);
+    _updateCustomChannelPermissions(roleId, channelId, permissions);
+  }
+
+  function removeCustomChannelPermissions(
```

Comment:
clear, not remove, and shouldn't need a string[] of permissions, as it will remove the override completely.